File: duplicates-android/app/src/main/java/com/github/duplicates/bookmark/BookmarkAdapter.java
/*
 * Source file of the Remove Duplicates project.
 * Copyright (c) 2016. All Rights Reserved.
 *
 * The contents of this file are subject to the Mozilla Public License Version
 * 2.0 (the "License"); you may not use this file except in compliance with the
 * License. You may obtain a copy of the License at
 * http://www.mozilla.org/MPL/2.0
 *
 * Contributors can be contacted by electronic mail via the project Web pages:
 *
 * https://github.com/pnemonic78/RemoveDuplicates
 *
 * Contributor(s):
 *   Moshe Waisberg
 *
 */
package com.github.duplicates.bookmark;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.github.android.removeduplicates.R;
import com.github.duplicates.DuplicateAdapter;

/**
 * List adapter for duplicate bookmarks.
 *
 * @author moshe.w
 */
public class BookmarkAdapter extends DuplicateAdapter<BookmarkItem, BookmarkViewHolder> {

    @Override
    public BookmarkViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.same_bookmark, parent, false);
        return new BookmarkViewHolder(itemView, this);
    }
}

File: duplicates-android/app/src/main/java/com/github/duplicates/bookmark/BookmarkAdapter.kt
/*
 * Copyright 2016, Moshe Waisberg
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.github.duplicates.bookmark

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import com.github.android.removeduplicates.databinding.SameBookmarkBinding
import com.github.duplicates.DuplicateAdapter

/**
 * List adapter for duplicate bookmarks.
 *
 * @author moshe.w
 */
class BookmarkAdapter : DuplicateAdapter<BookmarkItem, BookmarkViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BookmarkViewHolder {
        val context: Context = parent.context
        val binding = SameBookmarkBinding.inflate(LayoutInflater.from(context), parent, false)
        return BookmarkViewHolder(binding, this)
    }
}
